cnf-certification-test/chaostesting/suite.go

```diff
@@ -0,0 +1,280 @@
+	}
+	e = os.Remove("chaostesting/servicAccount.yaml.tmp")
+	if e != nil {
+		logrus.Debugf("error while removing the temp file of the servicAccount %e", e)
+	}
+	e = os.Remove("chaostesting/deleteExperment.yaml.tmp")
+	if e != nil {
+		logrus.Debugf("error while removing the temp file of the deleteExperment %e", e)
+	}
+}
+
+func applyTemplate(appLabel, appKind, namespace, fileename string) (string, error) {
+	input, err := os.ReadFile(fileename)
+	if err != nil {
+		fmt.Println(err)
+		return "", err
+	}
+	output := bytes.ReplaceAll(input, []byte("{{ APP_NAMESPACE }}"), []byte(namespace))
+	output = bytes.ReplaceAll(output, []byte("{{ APP_LABEL }}"), []byte(appLabel))
+	output = bytes.ReplaceAll(output, []byte("{{ APP_KIND }}"), []byte(appKind))
+	fileName := fileename + ".tmp"
+	if err = os.WriteFile(fileName, output, 0700); err != nil {
+		fmt.Println(err)
+		return "", err
+	}
+	return fileName, nil
+}
+
+func waitForTestFinish(timeout time.Duration) bool {
+	const pollingPeriod = 1 * time.Second
+	var elapsed time.Duration
+	var result bool
+	for elapsed < timeout {
+		result = waitForResult()
+
+		if result {
+			break
+		}
+		time.Sleep(pollingPeriod)
+		elapsed += pollingPeriod
+	}
+	return result
+}
+
+func returnResult() (bool, string) {
+	oc := clientsholder.GetClientsHolder()
+	gvr := schema.GroupVersionResource{Group: "litmuschaos.io", Version: "v1alpha1", Resource: "chaosresults"}
+	crs, err := oc.DynamicClient.Resource(gvr).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("error getting : %v\n", err)
+	}
+	for _, cr := range crs.Items {
+		result := cr.Object["status"].(map[string]interface{})["experimentStatus"].(map[string]interface{})["failStep"]
+		expKind := cr.Object["spec"].(map[string]interface{})["experiment"]
+		if expKind == chaosname {
+			if result == "N/A" {
+				return true, ""
+			}
+			return false, result.(string)
+
+		}
+	}
+	return false, ""
+}
+
+func waitForResult() bool {
+	oc := clientsholder.GetClientsHolder()
+	gvr := schema.GroupVersionResource{Group: "litmuschaos.io", Version: "v1alpha1", Resource: "chaosengines"}
+	crs, err := oc.DynamicClient.Resource(gvr).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("error getting : %v\n", err)
+	}
+	for _, cr := range crs.Items {
+		typ := cr.Object["status"].(map[string]interface{})["experiments"].([]interface{})
+		status := cr.Object["status"].(map[string]interface{})["engineStatus"]
+		if typ[0].(map[string]interface{})["name"] == chaosname {
+			if status == completedResult {
+				return true
+			} else {
+				return false
+			}
+		}
+	}
+	return false
+}
+
+//nolint:funlen //
+func createResource(filepath string) (bool, error) {
+	oc := clientsholder.GetClientsHolder()
+
+	b, err := os.ReadFile(filepath)
+	if err != nil {
+		return false, err
+	}
+	log.Printf("%q \n", string(b))
+
+	c := oc.K8sClient
+
+	dd := oc.DynamicClient
```

Comment:
Please use a meaningful variable names for the clients (c and dd). Suggestion: `typedClient` and `dynamicClient`.